app/services/katello/pulp3/repository/ansible_collection.rb

```diff
@@ -13,23 +13,27 @@ def client_class
         end
 
         def api_exception_class
+          self.class.api_exception_class
+        end
```

Comment:
this too can be moved to the base class repository.rb

doing some testing now!